src/hooks/wallets/web3.ts

```diff
@@ -14,16 +14,19 @@ export const getRpcServiceUrl = (rpcUri: RpcUri): string => {
   return formatRpcServiceUrl(rpcUri, INFURA_TOKEN)
 }
 
-export const createWeb3ReadOnly = ({ rpcUri }: ChainInfo): JsonRpcProvider => {
-  return new JsonRpcProvider({ url: getRpcServiceUrl(rpcUri), timeout: 10_000 })
+export const createWeb3ReadOnly = (rpcUri: RpcUri, customRpc?: string): JsonRpcProvider => {
+  return new JsonRpcProvider({ url: customRpc ? customRpc : getRpcServiceUrl(rpcUri), timeout: 10_000 })
 }
 
 export const createWeb3 = (walletProvider: EIP1193Provider): Web3Provider => {
   return new Web3Provider(walletProvider)
 }
 
-export const createSafeAppsWeb3Provider = ({ safeAppsRpcUri }: ChainInfo): JsonRpcProvider => {
-  return new JsonRpcProvider({ url: formatRpcServiceUrl(safeAppsRpcUri, SAFE_APPS_INFURA_TOKEN), timeout: 10_000 })
+export const createSafeAppsWeb3Provider = (safeAppsRpcUri: RpcUri, customRpc?: string): JsonRpcProvider => {
+  return new JsonRpcProvider({
+    url: customRpc ? customRpc : formatRpcServiceUrl(safeAppsRpcUri, SAFE_APPS_INFURA_TOKEN),
```

Comment:
```suggestion
    url: customRpc || formatRpcServiceUrl(safeAppsRpcUri, SAFE_APPS_INFURA_TOKEN),
```

Reply:
Unfortunately `||` will work with empty strings but in this case we want to fallback to `formatRpcServiceUrl` if `customRpc` is an empty string.